svm/tests/conformance.rs

```diff
@@ -0,0 +1,355 @@
+use {
+    crate::{
+        mock_bank::{MockBankCallback, MockForkGraph},
+        proto::InstrFixture,
+        transaction_builder::SanitizedTransactionBuilder,
+    },
+    lazy_static::lazy_static,
+    prost::Message,
+    solana_bpf_loader_program::syscalls::create_program_runtime_environment_v1,
+    solana_program_runtime::{
+        compute_budget::ComputeBudget,
+        loaded_programs::{ProgramCache, ProgramCacheEntry, ProgramRuntimeEnvironments},
+        solana_rbpf::{
+            program::{BuiltinProgram, FunctionRegistry},
+            vm::Config,
+        },
+        timings::ExecuteTimings,
+    },
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount, WritableAccount},
+        bpf_loader_upgradeable,
+        epoch_schedule::EpochSchedule,
+        feature_set::{FeatureSet, FEATURE_NAMES},
+        hash::Hash,
+        instruction::AccountMeta,
+        pubkey::Pubkey,
+        signature::Signature,
+    },
+    solana_svm::{
+        runtime_config::RuntimeConfig,
+        transaction_error_metrics::TransactionErrorMetrics,
+        transaction_processor::{ExecutionRecordingConfig, TransactionBatchProcessor},
+    },
+    std::{
+        collections::{HashMap, HashSet},
+        env,
+        ffi::OsString,
+        fs::{self, File},
+        io::Read,
+        path::PathBuf,
+        process::Command,
+        sync::{Arc, RwLock},
+    },
+};
+
+mod proto {
+    include!(concat!(env!("OUT_DIR"), "/org.solana.sealevel.v1.rs"));
+}
+mod mock_bank;
+mod transaction_builder;
+
+const fn feature_u64(feature: &Pubkey) -> u64 {
+    let feature_id = feature.to_bytes();
+    feature_id[0] as u64
+        | (feature_id[1] as u64) << 8
+        | (feature_id[2] as u64) << 16
+        | (feature_id[3] as u64) << 24
+        | (feature_id[4] as u64) << 32
+        | (feature_id[5] as u64) << 40
+        | (feature_id[6] as u64) << 48
+        | (feature_id[7] as u64) << 56
+}
+
+lazy_static! {
+    static ref INDEXED_FEATURES: HashMap<u64, Pubkey> = {
+        FEATURE_NAMES
+            .iter()
+            .map(|(pubkey, _)| (feature_u64(pubkey), *pubkey))
+            .collect()
+    };
+}
+
+fn setup() -> PathBuf {
+    let mut dir = env::current_dir().unwrap();
+    dir.push("test-vectors");
+    if !dir.exists() {
+        std::println!("Cloning test-vectors ...");
+        Command::new("git")
+            .args([
+                "clone",
+                "https://github.com/firedancer-io/test-vectors",
+                dir.as_os_str().to_str().unwrap(),
+            ])
+            .status()
+            .expect("Failed to download test-vectors");
+
+        std::println!("Checking out commit e94f22066cb2e98f8fe5f05a6797b8111ad78265");
+        Command::new("git")
+            .current_dir(&dir)
+            .args(["checkout", "e94f22066cb2e98f8fe5f05a6797b8111ad78265"])
```

Comment:
I'm thinking if we fork this repo to Anza then use it instead. I believe they won't rug us but they may break our test carelessly by force push/rename repo/delete repo ... 🤔 
cc @t-nelson

Reply:
I asked the firedancer team to block force pushes in the repository, so this may hedge against the commit disappearance. Also, I believe Github is smart enough to redirect requests to the correct repository if it gets renamed.

I prefer cloning from the original one, to reduce maintenance overhead (syncing the fork).

Reply:
We will be careful not to break your CI - we've disabled force pushes. Please let us know if you see any breakages and we'll prioritise fixing them ASAP.